File: leetcode/p0605/Solution.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int max = 0;
        
        for (int i = 0; i < flowerbed.length; ++i) {
            if (flowerbed[i] == 1) continue;
            
            int prev = (i > 0) ? flowerbed[i - 1] : 0;
            int next = (i < flowerbed.length - 1) ? flowerbed[i + 1] : 0;
            if (prev == 0 && next == 0) {
                flowerbed[i] = 1;
                max++;
            }
        }
        
        return n <= max;
    }
}

File: leetcode/p0605/Solution.kt
class Solution {
    fun canPlaceFlowers(flowerbed: IntArray, n: Int): Boolean {
        var max = 0
        
        for ((i, flower) in flowerbed.withIndex()) {
            if (flowerbed[i] == 1) {
                continue
            }
            val prev = if (i > 0) flowerbed[i - 1] else 0
            val next = if (i < flowerbed.size - 1) flowerbed[i + 1] else 0
            if (prev == 0 && next == 0) {
                flowerbed[i] = 1
                max++
            }
        }
        return n <= max
    }
}
